### PR TITLE
Data models and routing for "system overview" narratives

### LINKED ISSUE #275: Create data models for "system overview" narratives
**What needs to be done? Why does it need to be done?**
A "system overview" narrative is analogous to what we called a "DetailPage" in ND Spotlight. It follows a predictable structure, with an introductory title and description followed by a number of sections that each have their own title and description along with a visualization component that consumes metric data. There are currently four of them, for Sentencing, Prison, Probation, and Parole.

All of the copy for these pages is static, so it should be added to the content source. A data model that consumes this content should be created, and a mapping of these model instances should be added to the Tenant when it is instantiated. 

It is a feature of this particular Narrative, but not necessarily all Narratives, that all of the data it displays is represented by standalone Metrics. Therefore rather than fetching any data for itself, it should have relationships to the relevant Metric instances.

**Additional context**
Some details remain TBD at this point pending further design iterations. 
- will any (or all) metrics be displayed differently here than they would be on their standalone pages? Would they require supplemental data in any cases (e.g. if we are showing population maps here but not on the metric pages)? If so this new data model would be responsible for providing that
- There are also Collection models (which can be thought of as "tags" or "topics") that contain the exact same metrics as each of these narratives. Not sure yet if we should just consider that a coincidence, or if there will be some tangible benefit from creating a relationship between the two? Should be considered more deeply as part of this ticket

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

This creates the content and models for the "system overview" narratives, which are analogous to the main pages from Spotlight v1, and plugs them into the routing and navigation system. 

Now that the scaffolding is all in place, this was pretty straightforward! Note that most of the content I had originally assigned to the Metrics and Collections is really more appropriate for these narratives (which did not really exist in the product requirements until about a week ago), so I mostly just moved that over. I tried to replace the metric titles with what was in the data portal mocks and sub in something sensible if there was nothing there, but none of that should be considered final at this point. (all of those "description" fields are now just placeholders because I don't have anything to put there, and I am frankly not even sure if we will wind up needing those fields at all; they don't appear in the latest iterations of the mocks, but we can worry about that later). 

In the future we will have to disambiguate between different kinds of narratives, which is why the routing-related code refers to "narrativeTypeId" more generically; within the data models though, we will want to keep them separate to make UI treatment easier; different groups of narratives are not likely to resemble one another in structure or presentation.

Also note that I renamed "PageNarrativeHome" to "PageNarrativeList", which seems to have confused the Github diffing elves. Sorry! 🧝🏻‍♂️ 

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #275 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Review by jovergaag — COMMENTED** [review]
My big question here is around Narrative/Collection The models are quite different, but where and how they are used is so similar. Clicking 'Collections' in the navbar routes to `narratives`. As someone not knowing the future product well enough to know what the content of these pages will be, I have found it pretty confusing. I'm guessing that the `Collections` data will be interwoven into the `Narratives` copy? But why not call the route `collections` to match the nav?  

What am I missing that explains why these don't line up?

File: spotlight-client/src/routerUtils/getUrlForResource.ts [hunk 1]
```diff
@@ -50,9 +54,12 @@ function getUrlForResource(opts: GetUrlOptions): string {
       return `/${makeRouteParam(
         opts.params.tenantId
       )}/${DataPortalSlug}/${makeRouteParam(opts.params.metricTypeId)}`;
-    case "narratives":
+    case "narrative list":
```
[the remark this answers is not in the document]
Comment (macfarlandian) [reply spotlight-client/src/routerUtils/getUrlForResource.ts:57]: yeah it is very confusing and silly. "collections" used to be the word for what might properly be called "topics" or "tags" in the context of the data portal. then at some point in the latest round of designs it also got adopted as the "display title" of the page that we still refer to internally as "narratives", but they haven't really settled on that as the final title (it might be "narratives", it might be "stories", it might be something else) ... so I was trying to avoid using it in too many places. Typing that out, though, I can see that it doesn't make that much sense! At the very least I should have the title and the URL match up (and if they change later then they both change). But I think the names of components and stuff should be more stable than the display values, to avoid pointless churn, unless they drift so far apart as to become an impediment, so I would leave those as "narrative". Words! 💀 

More fun facts: I don't think there will actually be any relationship between the narratives and collections. They just so happen to completely overlap at the moment but that will not continue to be true. Also, I learned yesterday that the entire data portal feature is getting de-prioritized out of MVP scope, so the "explore" tab is going to come out of the navigation and get put on ice for a while. I may just remove the "collection" data model entirely so it doesn't turn into cruft, since it is not doing anything useful yet anyway.

File: spotlight-client/src/contentModels/SystemNarrative.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+import { SystemNarrativeContent } from "../contentApi/types";
```
[the remark this answers is not in the document]
Comment (macfarlandian) [reply spotlight-client/src/contentModels/SystemNarrative.ts:18]: that is tested as part of the `createTenant` tests, which integrate this module; the `tenant has system narratives` test verifies that the instantiated narratives correspond to what was in the content blob. I didn't really see any extra edge cases that seemed to warrant additional tests on this factory specifically

File: spotlight-client/src/PageNarrative/PageNarrative.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import { SystemNarrativeTypeId } from "../contentApi/types";
+import { useDataStore } from "../StoreProvider";
+import withRouteSync from "../withRouteSync";
+
+type PageNarrativeProps = RouteComponentProps & {
+  narrativeTypeId?: SystemNarrativeTypeId;
+};
+
+const PageNarrative: React.FC<PageNarrativeProps> = ({ narrativeTypeId }) => {
+  const tenant = useDataStore().tenantStore.currentTenant;
```
[the remark this answers is not in the document]
Comment (macfarlandian) [reply spotlight-client/src/PageNarrative/PageNarrative.tsx:30]: meaning, like, it would be `useDataStore().currentTenant` instead of this? yeah maybe that's a good idea ... I keep writing this line over and over again

**Comment by macfarlandian** [comment]
@jovergaag i think i would rather not mess with the data portal and "collections" stuff any further as part of this PR since it's not directly related to this functionality. your point is well taken though about what a general mess it is and I will aim to clean it up in the near future

File: spotlight-client/src/PageNarrative/PageNarrative.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jovergaag) [reply spotlight-client/src/PageNarrative/PageNarrative.tsx]: Or `useTenantStore().currentTenant`

File: spotlight-client/src/PageMetric/PageMetric.tsx [hunk 4]
```diff
@@ -25,7 +25,7 @@ import withRouteSync from "../withRouteSync";
 type PageMetricProps = RouteComponentProps & { metricTypeId?: MetricTypeId };
 
 const PageMetric: React.FC<PageMetricProps> = ({ metricTypeId }) => {
-  const tenant = useDataStore().tenantStore.currentTenant;
+  const { tenant } = useDataStore();
```
Comment (jovergaag) [inline spotlight-client/src/PageMetric/PageMetric.tsx:28]: Oh ya thats nicer.

**Review by jovergaag — APPROVED** [review]
Rock on 🎸

File: spotlight-client/src/PageNarrative/PageNarrative.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (macfarlandian) [reply spotlight-client/src/PageNarrative/PageNarrative.tsx]: oh i see what you meant .... I _think_ it is simpler for there to just be one context that contains the entire data store? that is sort of what they recommend but maybe it is not strictly necessary ... i haven't tried that
